src/components/forms/JurisdictionMetadata/index.tsx

```diff
@@ -0,0 +1,230 @@
+  redirectAfterAction: string;
+}
+
+export const defaultInitialValues: JurisdictionMetadataFormFields = {
+  file: new File([], ''),
+};
+
+// create payloads for sending
+const createPayloads = (result: any): SettingConfiguration[] => {
+  const payloads: SettingConfiguration[] = [];
+  const { data } = result;
+  // check if jurisdiction_id exists
+  if (data.length > 0 && data[0].jurisdiction_id) {
+    // get the metadata items
+    const headers = Object.keys(data[0]);
+    for (const header of headers) {
+      const settings: Setting[] = [];
+      if (header !== JURISDICTION_ID && header !== JURISDICTION_NAME) {
+        // add the metadata values with jurisdiction as the key
+        for (const item of data) {
+          const entries = Object.entries(item);
+          for (const [key, value] of entries) {
+            if (key === header) {
+              const setting: Setting = {
+                description: `${JURISDICTION_METADATA} for ${item.jurisdiction_name} id ${item.jurisdiction_id}`,
+                key: item.jurisdiction_id,
+                label: `${item.jurisdiction_name} metadata`,
+                value,
+              };
+              settings.push(setting);
+            }
+          }
+        }
+        const payload: SettingConfiguration = {
+          identifier: `jurisdiction_metadata-${header}`,
+          locationId: '',
+          providerId: 'demo',
+          settings,
+          teamId: '',
+          type: 'SettingConfiguration',
+        };
+        payloads.push(payload);
+      }
+    }
+  }
+  return payloads;
+};
+
+// read csv and convert to json
+const handleFile = (file: File, complete: (results: any) => any) => {
+  Papaparse.parse(file, {
+    complete: results => {
+      complete(results);
+    },
+    header: true,
+  });
+};
+
+export const submitForm = (
+  setSubmitting: (isSubmitting: boolean) => void,
+  setGlobalError: (errorMessage: string) => void,
+  setIfDoneHere: (closeSubmissionCycle: boolean) => void,
+  props: JurisdictionMetadataFormProps,
+  values?: JurisdictionMetadataFormFields
+) => {
+  if (values) {
+    const jurisdictionService = new props.OpenSRPService(OPENSRP_V1_SETTINGS_ENDPOINT);
+    handleFile(values.file, results => {
+      const payloads: SettingConfiguration[] = createPayloads(results);
+      if (payloads.length > 0) {
+        // loop through payloads and send
+        for (const payload of payloads) {
+          const valuesToSend = JSON.stringify(payload);
+          jurisdictionService
+            .create(valuesToSend)
+            .then(() => {
+              growl(FILE_UPLOADED_SUCCESSFULLY, {
+                onClose: () => setIfDoneHere(true),
+                type: toast.TYPE.SUCCESS,
+              });
+            })
+            .catch((e: Error) => {
+              setGlobalError(e.message);
+            });
+        }
+        setSubmitting(false);
+      } else {
+        setGlobalError('Invalid CSV');
```

Comment:
Maybe we should consider using the `displayError` utility?

Reply:
Maybe not because this validation happens locally

Reply:
Okay.  In that case then we should probably unset setGlobalError when you try to resubmit a form so that the error is not stuck on the screen, yes?

Reply:
I've unset it when a new file is selected